xbmc/cores/VideoPlayer/Process/VideoBuffer.cpp

```diff
@@ -0,0 +1,386 @@
+/*
+ *      Copyright (C) 2005-2017 Team XBMC
+ *      http://xbmc.org
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with XBMC; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "VideoBuffer.h"
+#include "threads/SingleLock.h"
+#include <string.h>
+
+//-----------------------------------------------------------------------------
+// CVideoBuffer
+//-----------------------------------------------------------------------------
+
+CVideoBuffer::CVideoBuffer(int id)
+{
+  m_id = id;
+  m_refCount = 0;
+}
+
+void CVideoBuffer::Acquire()
+{
+  m_refCount++;
+}
+
+void CVideoBuffer::Acquire(std::shared_ptr<IVideoBufferPool> pool)
+{
+  m_refCount++;
+  m_pool = pool;
+}
+
+void CVideoBuffer::Release()
+{
+  m_refCount--;
+  if (m_refCount <= 0)
+  {
+    m_pool->Return(m_id);
+    m_pool = nullptr;
+  }
+}
+
+AVPixelFormat CVideoBuffer::GetFormat()
+{
+  return m_pixFormat;
+}
+
+bool CVideoBuffer::CopyPicture(YuvImage* pDst, YuvImage *pSrc)
+{
+  uint8_t *s = pSrc->plane[0];
+  uint8_t *d = pDst->plane[0];
+  int w = pDst->width * pDst->bpp;
+  int h = pDst->height;
+  if ((w == pSrc->stride[0]) && ((unsigned int) pSrc->stride[0] == pDst->stride[0]))
```

Comment:
I just copied the old code: https://github.com/xbmc/xbmc/blob/master/xbmc/cores/VideoPlayer/DVDCodecs/DVDCodecUtils.cpp#L118

right, cast is not needed anymore